src/drivers/block_dev/ramdisk/ramdisk.c

```diff
@@ -167,7 +166,7 @@ static int read_sectors(block_dev_t *bdev,
 	char *read_addr;
 
 	ramdisk = (ramdisk_t *) bdev->privdata;
-	read_addr = ramdisk->p_start_addr + (blkno * ramdisk->block_size);
+	read_addr = ramdisk->p_start_addr + (blkno * bdev->block_size);
```

Comment:
It's already done
https://github.com/embox/embox/commit/6cc1c27f369e1d9609e6c69f816f6de54541206c